optimize/dataio.py

```diff
@@ -13,7 +13,7 @@ def structured_from_torch(tracks_torch, dtype):
     return rfn.unstructured_to_structured(tracks_torch.cpu().numpy(), dtype=dtype)
 
 class TracksDataset(Dataset):
-    def __init__(self, filename, ntrack, swap_xz=True, seed=3, random_ntrack=False, track_zlen_sel=2., 
+    def __init__(self, filename, ntrack, max_nbatch, iterations, swap_xz=True, seed=3, random_ntrack=False, track_zlen_sel=2., 
```

Comment:
Also style thing, but I would make `max_nbatch` a kwarg and default it to no max